src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -684,6 +733,38 @@ public boolean isPrivate() throws IOException {
         return getRepository().isPrivate();
     }
 
+    private <V> List<V> getResources(UriTemplate template, Class<? extends PagedApiResponse<V>> clazz) throws IOException, InterruptedException {
+        template.set("start", 0).set("limit", DEFAULT_PAGE_LIMIT);
+        String url = template.expand();
+
+        try {
+            List<V> resources = new ArrayList<>();
+            String response = getRequest(url);
+            PagedApiResponse<V> page = JsonParser.toJava(response, clazz);
+            resources.addAll(page.getValues());
+            while (!page.isLastPage()) {
```

Comment:
This was changed to a do, while in upstream and I would like to see that return 😅